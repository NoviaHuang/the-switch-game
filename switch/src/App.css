@media screen and (min-width: 1920px) {
  .home {
    text-align: center;
    background-color: #282c34;
    height: 100vh;
    overflow: hidden;
  }

  .home-header {
    position: relative;
    font-family: 'Courier New', 'Courier', 'monospace';
    font-size: 80px;
    color: white;
    right: 330px;
    bottom: 85px;
  }

  .home-img {
    position: relative;
    width: 40%;
    height: 55%;
    left: 450px;
    top: 100px;
  }

  .login-button {
    background-color: #e7e7e7;
    color: black;
    border-radius: 8px;
    border-color: #e7e7e7;
    padding: 10px 40px;
    font-size: 20px;
    margin: 10px;
    position: relative;
    bottom: 10px;
    -webkit-transition-duration: 0.4s;
    transition-duration: 0.4s;
  }

  .login-button:hover {
    background-color:rgb(93, 106, 134);
    color: white;
    border-color:rgb(93, 106, 134);
  }
}

@media screen and (max-width: 2140px) {
  .home {
    text-align: center;
    background-color: #282c34;
    height: 100vh;
    overflow: hidden;
  }

  .home-header {
    position: relative;
    font-family: 'Courier New', 'Courier', 'monospace';
    font-size: 80px;
    color: white;
    right: 330px;
    bottom: 85px;
  }

  .home-img {
    position: relative;
    width: 40%;
    height: 55%;
    left: 450px;
    top: 100px;
  }

  .login-button {
    font-family: 'Courier New', 'Courier', 'monospace';
    background-color: #e7e7e7;
    color: black;
    border-radius: 8px;
    border-color: #e7e7e7;
    padding: 10px 40px;
    font-size: 20px;
    margin: 10px;
    position: relative;
    bottom: 10px;
    -webkit-transition-duration: 0.4s;
    transition-duration: 0.4s;
  }

  .login-button:hover {
    background-color:rgb(93, 106, 134);
    color: white;
    border-color:rgb(93, 106, 134);
  }
}
