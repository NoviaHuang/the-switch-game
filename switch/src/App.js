import React, { Component } from 'react';
import './App.css'
import Login from './pages/LoginPage';
import bg from './background.png'

class App extends Component {
    constructor(){
        super();
        this.state={
            clicked: false
        };
        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(){
        this.setState({
            clicked: true
        });
    }

    render() {
        return (
            <div className="App">
              <img src={bg} className="App-bg" alt="background"/>
              <p className="App-header">WELCOME TO SWITCH</p>
              <div>
                <button onClick={this.handleClick}>Login</button>
                {this.state.clicked ? <Login /> : null}
              </div>
            </div>
        )
    }
}

export default App;
