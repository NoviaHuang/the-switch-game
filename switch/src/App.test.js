import 'react-testing-library/cleanup-after-each';
import 'jest-dom/extend-expect';
import React from 'react';
import App from './App';
import {render} from 'react-testing-library'
const puppeteer = require('puppeteer');

test("login successfully", async()=>{
  const browser = await puppeteer.launch({
        headless: false,
        slowMo: 80,
        args: ['--window-size=1920,1080']
      });
  const page = await browser.newPage();
  await page.goto('http://localhost:3000');
  await page.click("button.login-button");
  await page.url('http://localhost:3000/room-list');
  browser.close();
});