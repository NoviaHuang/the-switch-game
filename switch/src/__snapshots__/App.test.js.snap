// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page screenshot test Job Description page should render correctly 1`] = `
<Route>
  [Function]
</Route>
`;

exports[`Login Page screenshot test Login Page page should render correctly 1`] = `
<Route>
  [Function]
</Route>
`;
